lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -802,22 +1410,16 @@ Future<HttpFetchResponse> httpFetch(String url) async {
   }
 }
 
-Future<_DomResponse> _rawHttpPost(String url, String data) {
-  return js_util.promiseToFuture<_DomResponse>(js_util.callMethod(
-    domWindow,
-    'fetch',
-    <Object?>[
-      url,
-      js_util.jsify(<String, Object?>{
-        'method': 'POST',
-        'headers': <String, Object?>{
-          'Content-Type': 'text/plain',
-        },
-        'body': data,
-      }),
-    ],
-  ));
-}
+Future<_DomResponse> _rawHttpPost(String url, String data) =>
+    js_util.promiseToFuture<_DomResponse>(domWindow._fetch2(
+        url.toJS,
+        <String, Object?>{
+          'method': 'POST',
+          'headers': <String, Object?>{
+            'Content-Type': 'text/plain',
+          },
+          'body': data,
+        }.toJSAnyDeep));
```

Comment:
hah, I can see how this might be very confusing, because it is confusing :(.

`toJSAnyShallow` only converts a few type, notably primitives, typed data, and `List`s, but not, for example, `Map`s. For those cases we have to go deep.

Reply:
Aha, to me, "depth" meant... "recursive" transform, rather than a "shallow" transform (which only transforms "the first level"). I guess I associated it with shallow/deep cloning.

I don't have a better name for this, though :/